polygon/sync/block_downloader.go

```diff
@@ -149,7 +149,7 @@ func (d *blockDownloader) downloadBlocksUsingWaypoints(ctx context.Context, wayp
 				"sleepSeconds", d.notEnoughPeersBackOffDuration.Seconds(),
 			)
 
-			time.Sleep(d.notEnoughPeersBackOffDuration)
+			common.Sleep(ctx, d.notEnoughPeersBackOffDuration)
```

Comment:
are you sure that don't need to check `err`?

Reply:
common.Sleep does not return an err

Reply:
seems like you and @mh0lt may be suggesting that it should be returning an error if the parent ctx timeout/got cancelled so the caller knows - is this right?

Reply:
ok talked on chat with Mark so will make changes to:
- return err from common.Sleep coming from parentCtx.Err (once it is done)
- use time.NewTimer for the sleep time since it plays better with parentCtx.Done (timer gives us a channel to block on like ctx.Done)